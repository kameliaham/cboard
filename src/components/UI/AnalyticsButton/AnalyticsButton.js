import React from 'react';
import PropTypes from 'prop-types';
import { injectIntl, intlShape } from 'react-intl';
import AnalyticsIcon from '@material-ui/icons/BarChart';

import IconButton from '../IconButton';
import messages from '../../Analytics/Analytics.messages';

const propTypes = {
  /**
   * The component used for the root node. Either a string to use a DOM element or a component.
   */
  component: PropTypes.object,
  /**
   * If true, back button is disabled
   */
  disabled: PropTypes.bool,
  /**
   * @ignore
   */
  intl: intlShape.isRequired,
  /**
   * Callback fired when back button is clicked
   */
  onClick: PropTypes.func
};

function AnalyticsButton(props) {
  const { intl, ...other } = props;
  const label = intl.formatMessage(messages.analytics);

  return (
    <IconButton label={label} {...other}>
      <AnalyticsIcon />
    </IconButton>
  );
}

AnalyticsButton.propTypes = propTypes;

export default injectIntl(AnalyticsButton);
