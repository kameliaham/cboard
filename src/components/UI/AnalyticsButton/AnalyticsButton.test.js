import React from 'react';

import { matchSnapshotWithIntlProvider } from '../../../common/test_utils';
import AnalyticsButton from './AnalyticsButton';

describe('AnalyticsButton tests', () => {
  const props = {
    disabled: false,
    onClick: () => {}
  };

  test('default renderer', () => {
    matchSnapshotWithIntlProvider(<AnalyticsButton {...props} />);
  });
});
