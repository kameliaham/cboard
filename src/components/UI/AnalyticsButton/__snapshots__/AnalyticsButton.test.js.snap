// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnalyticsButton tests default renderer 1`] = `null`;
